tracing-appender/src/rolling.rs

```diff
@@ -558,17 +566,90 @@ impl Inner {
                     .unwrap_or(0),
             ),
             rotation,
+            keep_last: None,
         };
         Ok((inner, writer))
     }
 
+    fn prune_old_logs(&self, keep_last: usize) {
+        let format;
+        match self.rotation {
+            Rotation::MINUTELY => {
+                format = format_description::parse("[year]-[month]-[day]-[hour]-[minute]")
+                    .expect("Unable to create a formatter; this is a bug in tracing-appender");
+            }
+            Rotation::HOURLY => {
+                format = format_description::parse("[year]-[month]-[day]-[hour]")
+                    .expect("Unable to create a formatter; this is a bug in tracing-appender");
+            }
+            Rotation::DAILY => {
+                format = format_description::parse("[year]-[month]-[day]")
+                    .expect("Unable to create a formatter; this is a bug in tracing-appender");
+            }
+            Rotation::NEVER => {
+                unreachable!("keep last N files feature is not available for `Rotation::NEVER`!")
+            }
+        }
```

Comment:
if you take my suggestion about using the file's creation timestamp rather than parsing the date from the filename, we can remove this:
```suggestion
```